### PR TITLE
ci: Use FOSSA in dependency review workflow

### LINKED ISSUE #6393: Enable dependency license scanning in CI
This issue tracks the migration of the functionality provided by the GitLab CI job `license_scanning`

### PR DESCRIPTION
**Pull request by WillDaSilva**
Closes #6393

### CODE REVIEW

**Comment by WillDaSilva** [comment]
@tayloramurphy @DouweM could you please review the FOSSA report at https://app.fossa.com/projects/custom%2b32406%2fgithub.com%2fmeltano%2fmeltano/refs/branch/6423%2fmerge/6292392c8423e6255a44806e1f07c9c8a29def36, and resolve the alerts for licenses that are acceptable. For the licenses that are unacceptable, we'll have to find ways to remove the dependencies that introduce them.

I've gone ahead and resolve the alters for licenses that seem acceptable, and for the cases where FOSSA misidentified licenses (e.g. detecting a GPL license within some example code or docs that aren't part of the library itself).

**Comment by WillDaSilva** [comment]
The official FOSSA GitHub action doesn't provide a link to the report in the step that reports the error (`fossa test`), and it doesn't use a fixed version of `fossa-cli`, so I'm going to make a custom action that uses `fossa-cli`.

**Comment by WillDaSilva** [comment]
This same workflow is also being added to the SDK: https://github.com/meltano/sdk/pull/831

File: .github/workflows/dependency-review.yml [hunk 1]
```diff
@@ -14,7 +17,14 @@ jobs:
...
+        uses: actions/checkout@v3.0.2
+
+      - name: GitHub dependency vulnerability check
         uses: actions/dependency-review-action@v2
+
+      - name: FOSSA dependency license check
+        run: |
+          curl --no-progress-meter -H 'Cache-Control: no-cache' https://raw.githubusercontent.com/fossas/fossa-cli/v${FOSSA_CLI_VERSION}/install-latest.sh | bash
+          fossa analyze --fossa-api-key ${{ secrets.MELTYBOT_FOSSA_API_KEY }} --revision ${{ github.sha }}
+          fossa test --fossa-api-key ${{ secrets.MELTYBOT_FOSSA_API_KEY }} --revision ${{ github.sha }}
```
Comment (aaronsteers) [inline .github/workflows/dependency-review.yml:30]: Looks like they have a github action: https://github.com/fossas/fossa-action

Apologies in advance if you already considered this.

**Comment by aaronsteers** [comment]
> The official FOSSA GitHub action doesn't provide a link to the report in the step that reports the error (`fossa test`), and it doesn't use a fixed version of `fossa-cli`, so I'm going to make a custom action that uses `fossa-cli`.

Before reading this, I posted and deleted a comment asking if you'd considered their github action.

File: .github/workflows/dependency-review.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (WillDaSilva) [reply .github/workflows/dependency-review.yml:30]: Absolutely. TIL about `$GITHUB_STEP_SUMMARY`

[the remark this answers is not in the document]
Comment (WillDaSilva) [reply .github/workflows/dependency-review.yml:34]: image

The link is: https://app.fossa.com/projects/custom%2b32406%2fgithub.com%2fmeltano%2fmeltano/refs/branch/master/1a0529507172d3dc4da03e27808cb32bd2f3eef3

**Review by edgarrmondragon — APPROVED** [review]

**Review by pandemicsyn — APPROVED** [review]

**Pull request merged**